pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2601,6 +2601,15 @@ type StorageScopeSpec struct {
 	StorageClassDeviceSets []StorageClassDeviceSet `json:"storageClassDeviceSets,omitempty"`
 	// +optional
 	Store OSDStore `json:"store,omitempty"`
+	// +optional
+	// RestartInterval defines the time for which the OSD pods, that failed with zero exit code, will sleep before restarting.
+	// This is useful in scenarios like OSD flapping, when OSDs are marked down more than 5 times in 600 seconds by Ceph.
+	// Preventing the OSD pods to restart immediately in such scenarios will prevent Rook from marking OSD as `up` and thus
+	// peering of the PGs mapped to the OSD.
+	// The interval defaults to 6 hours if no value is provided. User needs to manually restart the OSD pod if they manage to fix
+	// the underlying OSD flapping issue before the restart interval.
+	// Valid time units are "m" or "h". For example, 90m, 2h, 6h, etc.
+	RestartInterval string `json:"restartInterval"`
```

Comment:
```suggestion
	RestartInterval string `json:"restartInterval"`
```
do we also need to expose this setting to downstream, right?

Reply:
The idea is not to expose this downstream. Default behavior is to have a 6h sleep. Users can change the time in the spec. So no downstream changes will be needed.